### PR TITLE
file: fix setting relative paths for src for hard links

### LINKED ISSUE #55971: file state=hard fails with relative path, docs say it should pass
##### SUMMARY

The docs say that the file module accepts a relative path for `src`. But if you try that with `state=hard` it fails.

##### ISSUE TYPE

- Bug Report

##### COMPONENT NAME

file module

##### ANSIBLE VERSION

```
ansible 2.9.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/opentelco/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/opentelco/patches/ansible/lib/ansible
  executable location = /home/opentelco/patches/ansible/bin/ansible
  python version = 2.7.15rc1 (default, Nov 12 2018, 14:31:15) [GCC 7.3.0]

```

##### CONFIGURATION
`ansible-config dump --only-changed`: empty

##### OS / ENVIRONMENT

Ubuntu 18

##### STEPS TO REPRODUCE

```yaml
---
- hosts: localhost
  tasks:
     - name: create hard link
       file:
          dest: hard.txt
          src: file.txt
          state: hard
```

##### EXPECTED RESULTS

File `hard.txt` should be a hard link to `file.txt`.

##### ACTUAL RESULTS

```
TASK [create hard link] ***************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "src must be an absolute path"}

```

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes #55971

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

file

### CODE REVIEW

**Review by samdoran — COMMENTED** [review]
Should we add a test using a relative path with `state=hard`?

**Comment by mkrizek** [comment]
@samdoran :+1: Done.

File: test/integration/targets/file/tasks/main.yml [hunk 1]
```diff
@@ -588,12 +585,77 @@
...
+- block:
+  - name: Create a testing file
+    file:
+      dest: original_file.txt
+      state: touch
+
+  - name: Test relative path with state=hard
+    file:
+      src: original_file.txt
+      dest: hard_link_file.txt"
```
Comment (samdoran) [inline test/integration/targets/file/tasks/main.yml]: Is that a dangling quote?
```suggestion
      dest: hard_link_file.txt
```

**Review by samdoran — APPROVED** [review]

File: test/integration/targets/file/tasks/main.yml (hunk 1 above)
Comment (mkrizek) [reply test/integration/targets/file/tasks/main.yml]: Good catch, thanks.

**Pull request merged**